File: comguanmuonehours/src/main/java/com/guanmu/onehours/MyApplication.java
package com.guanmu.onehours;

import android.app.Application;
import android.database.sqlite.SQLiteDatabase;

public class MyApplication extends Application {
    public DaoSession dao;

    @Override
    public void onCreate() {
        super.onCreate();
        //初始化数据库
        SharePreferenceUtils.init(getApplicationContext());

        DaoMaster.DevOpenHelper helper = new DaoMaster.DevOpenHelper(getApplicationContext(), "knowledge-db", null);
        SQLiteDatabase db = helper.getWritableDatabase();
        DaoMaster daoMaster = new DaoMaster(db);
        dao = daoMaster.newSession();
    }
}

File: comguanmuonehours/src/main/java/com/guanmu/onehours/MyApplication.kt
package com.guanmu.onehours

import android.app.Application
import android.database.sqlite.SQLiteDatabase

class MyApplication : Application() {
    lateinit var dao: DaoSession

    override fun onCreate() {
        super.onCreate()
        //初始化数据库
        SharePreferenceUtils.init(applicationContext)

        val helper = DaoMaster.DevOpenHelper(applicationContext, "knowledge-db", null)
        val db = helper.writableDatabase
        val daoMaster = DaoMaster(db)
        dao = daoMaster.newSession()
    }
}